apps/studio/components/interfaces/Database/Replication/RowMenu.tsx

```diff
@@ -110,17 +131,56 @@ export const RowMenu = ({
           <Button type="default" className="px-1.5" icon={<MoreVertical />} />
         </DropdownMenuTrigger>
         <DropdownMenuContent side="bottom" align="end" className="w-52">
-          {pipelineEnabled ? (
-            <DropdownMenuItem className="space-x-2" onClick={onDisablePipeline}>
-              <Pause size={14} />
-              <p>Stop pipeline</p>
-            </DropdownMenuItem>
-          ) : (
-            <DropdownMenuItem className="space-x-2" onClick={onEnablePipeline}>
-              <Play size={14} />
-              <p>Start pipeline</p>
-            </DropdownMenuItem>
-          )}
+          {(() => {
+            // Disable actions during transient request states
+            if (requestStatus !== PipelineStatusRequestStatus.None) {
+              return (
+                <DropdownMenuItem className="space-x-2 opacity-50" disabled>
+                  <Ban size={14} />
+                  <p>Action unavailable</p>
+                </DropdownMenuItem>
+              )
+            }
+
+            let label = 'Action unavailable'
+            let icon = <Play size={14} />
+            let onClick: (() => void | Promise<void>) | undefined
+            let disabled = false
+
+            if (statusName === PipelineStatusName.STOPPED) {
+              label = 'Start pipeline'
+              icon = <Play size={14} />
+              onClick = () => {
+                void onEnablePipeline()
+              }
+            } else if (statusName === PipelineStatusName.STARTED) {
+              label = 'Stop pipeline'
+              icon = <Pause size={14} />
+              onClick = () => {
+                void onDisablePipeline()
+              }
+            } else if (statusName === PipelineStatusName.FAILED) {
+              label = 'Restart pipeline'
+              icon = <RotateCcw size={14} />
+              onClick = () => {
+                void onRestartPipeline()
+              }
+            } else {
+              disabled = !PIPELINE_ACTIONABLE_STATES.includes((statusName ?? '') as any)
+              icon = <Ban size={14} />
+            }
+
+            return (
+              <DropdownMenuItem
+                className={`space-x-2 ${disabled ? 'opacity-50 cursor-not-allowed' : ''}`}
+                disabled={disabled}
+                onClick={onClick}
+              >
+                {icon}
+                <p>{label}</p>
+              </DropdownMenuItem>
+            )
+          })()}
```

Comment:
ideally i'd like to avoid this way of rendering components from a consistency POV - we can declare the components outside with a `const`